spec/benchmark_spec.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe "Performance testing" do
+
+  let(:disposable_domain) { ValidEmail2.disposable_emails.first }
+
+  it "lookup timing" do
+    # preload disposable emails
+    list    = ValidEmail2.disposable_emails
+    address = ValidEmail2::Address.new("test@example.com")
+
+    expect { address.send(:domain_is_in?, list) }.to perform_under(0.0001).sample(10).times
```

Comment:
You shouldn't assume the performance of the testing system like this. If I run the tests on a potato (or any other system with high load, such as CI systems), this will fail.

It would be better to just test whether the returned value from `.whitelist` / `.blacklist` is to `be_kind_of Set`. You can then assume log access time on these objects.

Reply:
I rewrote the test to do a blackbox test and to check the performance of `disposable_domain?` directly. There are many things that can be fucked up, because the method wants to do other stuff in the future, like happened in the past.

Reply:
I added a check to test the kind of the list.

Reply:
> check the performance of `disposable_domain?` directly

By setting a hard limit on 0.1ms this won't get you anywhere because you're measuring execution time. Have a look at `perform_logarithmic` to check complexity.